### PR TITLE
Fredrik/add validation resource forms

### PR DESCRIPTION
**Pull request by fredrikolovsson**
This PR implements the following specs:

- Validation needed in tree data (only did "musts" + level for localities): https://docs.google.com/document/d/1jocCR1hQH-vfSlG4VuTTeAV6sa7-R0dGzZjgCo8BRQI/edit#
- Updated validation message for lat/long: https://docs.google.com/document/d/1w7ABzoefXsp-CxM5EC-9NR6MWs_WfbjlxGdi7Bz-ht4/edit#
- Improved messages (did everything): https://docs.google.com/document/d/1fsbxjKMbTFfLXJ98UQRWj_hnKtwEmy5-dMGTaBwDAFA/edit

### CODE REVIEW

File: packages/ui/src/domainModules/locality/components/LocalityManager/item/BaseForm/transformations/syncErrors/mapParentError.js [hunk 1]
```diff
@@ -0,0 +1,21 @@
+import objectPath from 'object-path'
+
+const mapParentError = formValues => {
```
Comment (idali0226) [inline packages/ui/src/domainModules/locality/components/LocalityManager/item/BaseForm/transformations/syncErrors/mapParentError.js]: Change formValues to syncErrors

File: packages/ui/src/domainModules/taxon/schemas.js [hunk 2]
```diff
@@ -3,14 +3,63 @@ import schemaInterface from 'common/es5/schemaInterface'
 
 const models = schemaInterface.getModels()
 
-const createTaxonFormModels = () => {
-  const updatedModels = immutable.set(models, 'taxon.properties.parentId', {
-    type: 'string',
+const createFormModels = () => {
+  let updatedModels = { ...models }
+  // const updatedModels = immutable.set(models, 'taxon.properties.parentId', {
```
Comment (idali0226) [inline packages/ui/src/domainModules/taxon/schemas.js]: Remove comments

File: packages/ui/src/domainModules/locality/components/LocalityManager/item/BaseForm/transformations/syncErrors/mapParentError.js (hunk 1 above)
Comment (fredrikolovsson) [reply packages/ui/src/domainModules/locality/components/LocalityManager/item/BaseForm/transformations/syncErrors/mapParentError.js]: 0d5d08a

File: packages/ui/src/domainModules/taxon/schemas.js (hunk 2 above)
Comment (fredrikolovsson) [reply packages/ui/src/domainModules/taxon/schemas.js]: 2308d5f

**Pull request merged**